flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/DefaultSchedulerTest.java

```diff
@@ -1131,6 +1131,51 @@ public void testExceptionHistoryWithRestartableFailure() {
                                 failingException, updateStateTriggeringJobFailureTimestamp)));
     }
 
+    @Test
+    public void testExceptionHistoryWithPreDeployFailure() {
+        final JobGraph jobGraph = singleNonParallelJobVertexJobGraph();
+
+        // disable auto-completing slot requests to simulate timeout
+        executionSlotAllocatorFactory
+                .getTestExecutionSlotAllocator()
+                .disableAutoCompletePendingRequests();
+        final DefaultScheduler scheduler = createSchedulerAndStartScheduling(jobGraph);
+
+        executionSlotAllocatorFactory.getTestExecutionSlotAllocator().timeoutPendingRequests();
+
+        final ArchivedExecutionVertex taskFailureExecutionVertex =
+                Iterables.getOnlyElement(
+                        scheduler
+                                .requestJob()
+                                .getArchivedExecutionGraph()
+                                .getAllExecutionVertices());
+
+        // pending slot request timeout triggers a NoResourceAvailableException
+        final NoResourceAvailableException noResourceAvailableException =
+                new NoResourceAvailableException();
+        final long updateStateTriggeringRestartTimestamp =
+                initiateFailure(
```

Comment:
why do we need to initiate the failure? Shouldn't this have already happened when the pending requests timed out?

Reply:
You're right. That was some leftover from my pre-investigation test implementation 👍